SConstruct

```diff
@@ -415,6 +415,27 @@ if windows:
 env = Environment(**envDict)
 del envDict
 
+# The Scons 'default' tool enables a lot of tools that we don't actually need to enable.
+# On platforms like Solaris, it actually does the wrong thing by enabling the sunstudio
+# toolchain first. As such it is simpler and more efficient to manually load the precise
+# set of tools we need for each platform.
+# If we aren't on a platform where we know the minimal set of tools, we fallback to loading
+# the 'default' tool.
+def decide_platform_toolchain():
+    if windows:
+        # we only support MS toolchain on windows
+        return ['msvc', 'mslink', 'mslib']
```

Comment:
we don't need it currently, but as we discussed in person, consumers of the library may need it.